module/user-activity/src/commonTest/kotlin/org/cru/godtools/shared/user/activity/model/UserActivityTest.kt

```diff
@@ -197,6 +198,17 @@ class UserActivityTest {
         }
     }
 
+    @Test
+    fun testUserActivityBadgesSorting() {
+        counters[TOOL_OPEN("kgp")] = 5
+        counters[ARTICLE_OPEN("https://example.com/a".toUriOrNull()!!)] = 5
+        counters[TIPS_COMPLETED] = 5
+
+        val badges = UserActivity(counters).badges
+        assertTrue(badges.take(3).all { it.isEarned })
+        assertTrue(badges.drop(3).none { it.isEarned })
```

Comment:
here's the test, the first 3 badges should be earned, the remaining badges should not be earned